services/minespace-web/src/components/Forms/projects/projectSummary/ProjectContacts.tsx

```diff
@@ -0,0 +1,284 @@
+                  label="Country"
+                  required={isPrimary}
+                  validate={isPrimary ? [required] : []}
+                  data={countryOptions}
+                  component={RenderSelect}
+                  onSelect={(e) =>
+                    handleClearProvince(e, address_type_code, sub_division_code, field)
+                  }
+                />
+              </Col>
+              <Col md={12} sm={24}>
+                <Field
+                  name={`${field}.address.sub_division_code`}
+                  label="Province"
+                  required={isPrimary && !isInternational}
+                  data={provinceOptions.filter((p) => p.subType === address_type_code)}
+                  validate={!isInternational && isPrimary ? [required] : []}
+                  component={RenderSelect}
+                />
+              </Col>
+            </Row>
+
+            <Row gutter={16}>
+              <Col md={12} sm={24}>
+                <Field
+                  name={`${field}.address.city`}
+                  label="City"
+                  required={isPrimary}
+                  validate={isPrimary ? [required] : []}
+                  component={RenderField}
+                />
+              </Col>
+              <Col md={12} sm={24}>
+                <Field
+                  name={`${field}.address.post_code`}
+                  label="Postal Code"
+                  component={RenderField}
+                  required={isPrimary}
+                  validate={[postalCodeWithCountry(address_type_code), maxLength(10)]}
+                />
+              </Col>
+            </Row>
+
+            {index === 0 && (
+              <>
+                <Typography.Title level={3}>
+                  Additional project contacts (optional)
+                </Typography.Title>
+                <Typography.Paragraph>
+                  Provide contact information for additional people we can contact about this
+                  project.
+                </Typography.Paragraph>
+              </>
+            )}
+          </div>
+        );
+      })}
+      <LinkButton
+        onClick={() => fields.push({ is_primary: false })}
+        title="Add additional project contacts"
+      >
+        <PlusOutlined /> Add additional project contacts
+      </LinkButton>
+    </>
+  );
+};
+
+export const ProjectContacts: FC<ContactsProps> = (props) => {
+  useEffect(() => {
+    if (isNil(props.contacts) || props.contacts.length === 0) {
+      props.arrayPush(FORM.ADD_EDIT_PROJECT_SUMMARY, "contacts", { is_primary: true });
+    }
+  }, []);
+
+  return (
+    <>
+      <Typography.Title level={3}>Project Contacts</Typography.Title>
+      <FieldArray name="contacts" countryOptions={props.countryOptions} component={contacts} />
+    </>
+  );
+};
+
+const selector = formValueSelector(FORM.ADD_EDIT_PROJECT_SUMMARY);
+const mapStateToProps = (state) => ({
+  contacts: selector(state, "contacts"),
```

Comment:
I see that above you've got contacts defined with a selector, so this one can go (and the selector on 271), and there's also dispatch defined from `useDispatch`.
So you can take out the contacts here because it's coming from the selector, and you could get rid of the mapDispatchToProps (and also take out arrayPush from props) by doing a dispatch(arrayPush(...)) instead. What the mapDispatchToProps does is have it automatically be dispatched as part of the function. Kind of confusing to do both though.

Then you'd be able to do export default ProjectContacts.

Reply:
Nice, I removed the mapDispatchToProps, the contacts selectors and went with dispatch(arrayPush(..)) as suggested.